lib/saml_config.rb

```diff
@@ -2,12 +2,10 @@
 
 module Saml
   class Config
-    def initialize
-      @@settings ||= build_settings.freeze
-    end
+    attr_reader :settings
 
-    def settings
-      @@settings
+    def initialize
+      @settings ||= build_settings.freeze
```

Comment:
you've changed this from a class variable to an instance variable, which means that the config will be re-read from disk on every instantiation. That's what the class variable was trying to avoid.

We've solve this rubocop issue with `cattr_accessor` in the idp app. Maybe use that as a guide?